File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    hash = {}
    
    for i in tangerine:
        if i in hash:
            hash[i] += 1
        else:
            hash[i] = 1
            
    sorted_hash = sorted(hash.values(), reverse=True)
            
    for j in sorted_hash:
        if k <= 0:
            break
        k = k - j
        answer += 1
    
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    let hash = {};
    for (let i = 0; i < tangerine.length; i++) {
        if (hash[tangerine[i]]) {
            hash[tangerine[i]]++;
        } else {
            hash[tangerine[i]] = 1;
        }
    
    }
    
    let sorted = Object.entries(hash).sort((a, b) => b[1] - a[1]);
    
    for (let i = 0; i < sorted.length; i++) {
        if (k <= 0) {
            break;
        }
        k = k - sorted[i][1];
        answer++;
    }
    
    return answer;
}